src/hooks/useSubscriptions.ts

```diff
@@ -15,19 +15,32 @@
 // along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
 
 import { ChainDetails } from '../types/sourceTargetTypes';
+import { Subscriptions } from '../types/subscriptionsTypes';
 import useBlocksInfo from './useBlocksInfo';
 import useBridgedBlocks from './useBridgedBlocks';
-import useDashboardProfile from './useDashboardProfile';
+import useChainProfile from './useChainProfile';
 import useMessagesLane from './useMessagesLane';
 
-const useDashboard = (ChainDetail: ChainDetails) => {
-  const { api, destination, local, isApiReady, polkadotjsUrl } = useDashboardProfile(ChainDetail);
+interface Source {
+  source: string;
+  polkadotjsUrl: string;
+}
 
-  const blockInfo = useBlocksInfo({ api, chain: local, isApiReady });
-  const bridgedBlocks = useBridgedBlocks({ api, chain: destination, isApiReady });
-  const messagesLane = useMessagesLane({ api, chain: destination, isApiReady });
+type Output = Subscriptions & Source;
 
-  return { ...blockInfo, ...bridgedBlocks, ...messagesLane, local, polkadotjsUrl };
+const useSubscriptions = (ChainDetail: ChainDetails): Output => {
+  const {
+    apiConnection: { api, isApiReady },
+    target,
+    source,
+    polkadotjsUrl
+  } = useChainProfile(ChainDetail);
+
+  const blockInfo = useBlocksInfo({ api, chain: source, isApiReady });
```

Comment:
I have never agree that this use is incorrect. A hook inside a hook is composability, so this will not be changed.

Reply:
As we have discussed before, my opinion is that it is a matter of simplicity - as soon as useMessagesLane, useBridgedBlocks, useBlocksInfo are not used anywhere else, I dont understand why they need to be hooks. In any case, as soon as you decided that this will not be changed - my point is moot.

Reply:
Composing hooks is definitely expected and desirable - we shouldn't be inlining huge amounts of code, especially if there is a re-use potential.

That said, there is a thin line between composing for improving readability and making the code more complex due to too many details being hidden.
In this particular case I think it looks good as-is.